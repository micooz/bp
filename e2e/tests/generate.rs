use bp_cli::{
    commands::generate,
    options::generate::{ConfigType, GenerateOptions},
};
use e2e::fs;

#[tokio::test(flavor = "multi_thread")]
async fn test_generate_client_config() {
    let config_path = "tests/tmp/client.config.json";

    generate::run(GenerateOptions {
        config: Some(config_path.to_string()),
        ..Default::default()
    })
    .await;

    insta::assert_json_snapshot!(fs::read_file(config_path).await);
}

#[tokio::test(flavor = "multi_thread")]
async fn test_generate_server_config() {
    let config_path = "tests/tmp/server.config.yaml";

    generate::run(GenerateOptions {
        config: Some(config_path.to_string()),
        config_type: ConfigType::Server,
        ..Default::default()
    })
    .await;

    insta::assert_yaml_snapshot!(fs::read_file(config_path).await);
}
