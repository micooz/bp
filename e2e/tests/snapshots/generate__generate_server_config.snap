---
source: e2e/tests/generate.rs
assertion_line: 31
expression: "fs::read_file(config_path).await"

---
"---\nbind: \"__some_where__:3000\"\nkey: __some_key__\nencryption: erp\ndns_server: \"8.8.8.8:53\"\ntls: false\nquic: false\ntls_cert: ~\ntls_key: ~\n"
